compiler/passes/normalize.cpp

```diff
@@ -750,6 +750,12 @@ static void insertCallTempsForRiSpecs(BaseAST* base) {
 
   for_vector(ForallStmt, fs, forallStmts) {
     for_shadow_vars(svar, temp, fs) {
+      if (svar->specBlock == nullptr)
+        continue;
+      // hoist the details, if any out of the ForallStmt
+      for (AList& sbody = svar->specBlock->body;
+           sbody.head != sbody.tail;
+           fs->insertBefore(sbody.head->remove()));
```

Comment:
Could you explain why this is necessary?

Reply:
Good question. Existing compiler code did not handle the case where the `specBlock` contained some type computations, like `2*int` in the case of `minmax(2*int)`. This is a small change to adjust the AST so it fits with the rest of the compiler / reduce intent machinery. It certainly does not need to be replicated in Dyno. :)

Fundamentally we need some place to "perform" these computations, at least for the purposes of resolution. (They will not have a runtime type because the outermost type cannot be an array or a domain.) Doing so before entering the forall loop proper looked like a good option to me.